internal/rsl/rsl.go

```diff
@@ -328,7 +327,20 @@ func GetLatestEntry(repo *git.Repository) (Entry, error) {
 // GetLatestNonGittufReferenceEntry returns the first reference entry that is
 // not for the gittuf namespace.
 func GetLatestNonGittufReferenceEntry(repo *git.Repository) (*ReferenceEntry, []*AnnotationEntry, error) {
-	it, err := GetLatestEntry(repo)
+	return GetLatestNonGittufReferenceEntryFrom(repo, plumbing.ZeroHash)
+}
+
+func GetLatestNonGittufReferenceEntryFrom(repo *git.Repository, anchor plumbing.Hash) (*ReferenceEntry, []*AnnotationEntry, error) {
+	var (
+		it  Entry
+		err error
+	)
+
+	if anchor.IsZero() {
+		it, err = GetLatestEntry(repo)
+	} else {
+		it, err = GetEntry(repo, anchor)
+	}
```

Comment:
This looks like an unrelated diff. Is this rebased on another change?

Reply:
rip, yes. sorry, will fix if we decide to proceed with this PR!